pkg/cluster/self_hosted.go

```diff
@@ -113,3 +114,93 @@ func (c *Cluster) newSelfHostedSeedMember() error {
 	c.logger.Infof("self-hosted cluster created with seed member (%s)", newMemberName)
 	return nil
 }
+
+func (c *Cluster) migrateBootMember() error {
+	endpoint := c.spec.SelfHosted.BootMemberClientEndpoint
+
+	c.logger.Infof("migrating boot member (%s)", endpoint)
+
+	cfg := clientv3.Config{
+		Endpoints:   []string{endpoint},
+		DialTimeout: constants.DefaultDialTimeout,
+	}
+	etcdcli, err := clientv3.New(cfg)
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), constants.DefaultRequestTimeout)
+	resp, err := etcdcli.MemberList(ctx)
+	cancel()
+	etcdcli.Close()
+	if err != nil {
+		return err
+	}
+
+	if len(resp.Members) != 1 {
+		return fmt.Errorf("boot cluster contains more than one member")
+	}
+	bootMember := resp.Members[0]
+	bootMemberID := resp.Members[0].ID
+
+	initialCluster := make([]string, 0)
+	for _, purl := range bootMember.PeerURLs {
+		initialCluster = append(initialCluster, fmt.Sprintf("%s=%s", bootMember.Name, purl))
+	}
+
+	// create the first member inside Kubernetes for migration
+	newMemberName := fmt.Sprintf("%s-%04d", c.name, c.idCounter)
+	c.idCounter++
+
+	peerURL := "http://$(MY_POD_IP):2380"
+	initialCluster = append(initialCluster, newMemberName+"="+peerURL)
```

Comment:
I think we should keep as it is now.

part1 is 4 lines of code and it is clear

part2 should be refactored with other funcs. basically we should hide these k8s calls when possible

part3 is 2 line of code and it is clear

part4 should be removed in the near future from this func once reconcile becomes a little smarter.